### PR TITLE
WIP: ensure saved prerendered assets are stored as file and not as a folder

### LINKED ISSUE #9560: prerendered assets are stored without file ending
### Describe the bug

When `prerender` set to `true` and a `+page.ts` file makes a `fetch` request, the response of that request get's written to disk.
But it does so without a file ending. So when making a request to `/api` and `/api/child` the first response get's saved as
`/api` (file) and when it tires to save the second response, an error get's thrown because `SvelteKit` tries to create the `/api` folder to save the `/api/child` file. 

### Reproduction

https://stackblitz.com/edit/sveltejs-kit-template-default-vrpjqd

Running `npm run build` will throw an error.

```
[Error [ENOTDIR]: ENOTDIR: not a directory, mkdir '/home/projects/sveltejs-kit-template-default-vrpjqd/.svelte-kit/output/prerendered/dependencies/api'] {
  syscall: 'mkdir',
  errno: -20,
  code: 'ENOTDIR',
  path: '/home/projects/sveltejs-kit-template-default-vrpjqd/.svelte-kit/output/prerendered/dependencies/api'
}
```

### Logs

_No response_

### System Info

```Shell
`StackBlitz`
```

### Severity

blocking an upgrade

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ivanhofer**
closes #9560

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/kit/src/core/postbuild/prerender.js [hunk 1]
```diff
@@ -132,14 +132,20 @@ async function prerender({ out, manifest_path, metadata, verbose, env }) {
...
-	function output_filename(path, is_html) {
+	function output_filename(path, is_html, type) {
 		const file = path.slice(config.paths.base.length + 1) || 'index.html';
 
 		if (is_html && !file.endsWith('.html')) {
 			return file + (file.endsWith('/') ? 'index.html' : '.html');
 		}
 
+		if (!file.includes('.')) {
+			const ext = type === 'application/json' ? 'json' : 'txt';
```
Comment (ivanhofer) [inline packages/kit/src/core/postbuild/prerender.js:145]: In the end the file ending does not really matter right?

**Comment by dummdidumm** [comment]
Does this really solve the problem or just shift it elsewhere? If you have `api/+server.js` and `api.json/+server.js` then there's a conflict with this PR.

**Comment by ivanhofer** [comment]
Yeah, I just realized that this does not relly fix the issue.. It must probably be a unique file ending (`.sveltekit_prerender` ?).
And also the code that references the generated file needs to know about this file ending or esle it tries to import a file that does not exists.

**Comment by ivanhofer** [comment]
Or use the `__data.json` approach like it already happens for `load` functions? Shouldn't prerendered `load` and `GET` be the same except that `GET` could return not just `json`?
So maybe rewriting `GET` responses always to `__data.txt`?

File: .changeset/witty-spies-sin.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+ensure saved prerendered assets are stored as file and not as a folder
```
Comment (benmccann) [inline .changeset/witty-spies-sin.md:5]: I haven't looked closely, but assuming this is a bug, we should use the `fix: ` prefix for consistency in the changelog https://github.com/sveltejs/kit/blob/master/packages/kit/CHANGELOG.md
```suggestion
fix: ensure saved prerendered assets are stored as file and not as a folder
```

**Comment by Rich-Harris** [comment]
closing per https://github.com/sveltejs/kit/issues/9560#issuecomment-1511969203

**Pull request closed without merging**